File: 532. K-diff Pairs in an Array.java
public class Solution {
    public int findPairs(int[] nums, int k) {
        if (nums.length == 0) {
            return 0;
        }
        Arrays.sort(nums);
        int cnt = 0, tmp = nums[0] - 1;
        if (k == 0) {
            for (int i = 0; i < nums.length - 1; tmp = nums[i], i++) {
                if (nums[i] == tmp) {
                    continue;
                }
                if (nums[i + 1] == nums[i]) {
                    cnt++;
                }
            }
        }
        if (k > 0) {
            for (int i = 0; i < nums.length - 1; tmp = nums[i], i++){
                if (tmp == nums[i]) {
                    continue;
                }
                for (int j = i + 1; j < nums.length; j++) {
                    if (tmp == nums[i]) {
                        continue;
                    }
                    if (nums[j] - nums[i] == k) {
                        cnt++;
                        break;
                    }
                    if (nums[j] - nums[i] > k) {
                        break;
                    }
                }
            }        
        }
        return cnt;
    }
}

File: 532. K-diff Pairs in an Array.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var findPairs = function(nums, k) {
    nums.sort(function(a, b) {return a - b})
    var cnt = 0, tmp = nums[0] - 1
    if (k == 0) {
        for (var i = 0; i < nums.length - 1; tmp = nums[i], i++) {
            if (nums[i] == tmp) {
                continue
            }
            if (nums[i + 1] == nums[i]) {
                cnt++
            }
        }
    }
    if (k > 0) {
        for (i = 0, tmp; i < nums.length - 1; tmp = nums[i], i++) {
            if (nums[i] == tmp) {
                continue
            }
            for (var j = 0; j < nums.length; j++) {
                if (nums[j] - nums[i] < k) {
                    continue
                }
                if (nums[j] - nums[i] == k) {
                    cnt++
                    break
                }
                if (nums[j] - nums[i] > k) {
                    break
                }
            }
        }
    }
    return cnt
};
